opendcs-rest-api/src/main/java/org/opendcs/odcsapi/sec/cwms/CwmsAuthorizationDAO.java

```diff
@@ -44,11 +44,18 @@ public Set<OpenDcsApiRoles> getRoles(String username) throws DbException
 	{
 		Set<OpenDcsApiRoles> roles = EnumSet.noneOf(OpenDcsApiRoles.class);
 		roles.add(OpenDcsApiRoles.ODCS_API_GUEST);
-		String q = "SELECT user_group_id" +
-				" FROM av_sec_users" +
-				" WHERE db_office_code = cwms_util.get_db_office_code(?)" +
-				" AND username = ?" +
-				" AND is_member = 'T'";
+		String q = "select user_group_id " +
+				"from cwms_20.av_sec_users " +
+				"where db_office_code = cwms_20.cwms_util.get_db_office_code(:input_db_office_code) " +
+				"  and upper(username) = case " +
+				"     when instr(:username_str, '.', -1) > 0 then " +
+				"        (select userid " +
+				"           from cwms_20.at_sec_cwms_users " +
+				"           where edipi = substr(:username_str, instr(:username_str, '.', -1) + 1)) " +
```

Comment:
I think you need office id in this query on at_sec_cwms_users

Reply:
That office column doesn't look to be for the CWMS office. It isn't foreign keyed and is initialized to null for all test users.

Reply:
Check the equivalent CDA code. The roles returned *should* be tied to a specific office for the user. If there's a foreign key missing I think that might a bug in the schema.

Reply:
Userid is primary key

Reply:
`av_sec_users` is the one with the office code, `at_sec_cwms_users` has an 'office' column looks to be used for something else (it's null in our database exports and 16 bytes).